### PR TITLE
Sql sanitizer: handle double quoted table names

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/5691

### CODE REVIEW

File: instrumentation-api/src/main/jflex/SqlSanitizer.jflex [hunk 1]
```diff
@@ -372,6 +372,13 @@ WHITESPACE        = [ \t\r\n]+
           if (dialect == SqlDialect.COUCHBASE) {
             builder.append('?');
           } else {
+            if (!insideComment && !extractionDone) {
+              extractionDone = operation.handleIdentifier();
+              // table names extracted here will be surrounded with double quotes
+              if (operation.mainTable != null && operation.mainTable.startsWith("\"") && operation.mainTable.endsWith("\"")) {
+                operation.mainTable = operation.mainTable.substring(1, operation.mainTable.length() - 1);
+              }
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/main/jflex/SqlSanitizer.jflex:380]: Nit: WDYT about moving this `if` into the `handleIdentifier()` method?

File: instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy [hunk 2]
```diff
@@ -105,6 +105,8 @@ class SqlStatementSanitizerTest extends Specification {
     sql                                                               | expected
     // Select
     'SELECT x, y, z FROM schema.table'                                | SqlStatementInfo.create(sql, 'SELECT', 'schema.table')
+    'SELECT x, y, z FROM `schema.table`'                              | SqlStatementInfo.create(sql, 'SELECT', 'schema.table')
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy:108]: Wait, did we have backtick ` parsing?

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy (hunk 2 above)
Comment (laurit) [reply instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy:108]: Yes, this already worked just didn't have a test.

Comment (mateuszrzeszutek) [reply instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy:108]: Nice 👍

Comment (laurit) [reply instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy:108]: Actually it only works fine when you don't have a space inside backticks. If there is a space it returns the first word as the table name.

Comment (mateuszrzeszutek) [reply instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy:108]: Hm, we don't have any special rules for backticks, I guess they're just treated as "other characters" - that maybe that behavior is caused by this?

Comment (laurit) [reply instrumentation-api/src/test/groovy/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizerTest.groovy]: yes, hopefully fixed now.

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation-api/src/main/jflex/SqlSanitizer.jflex (hunk 1 above)
Comment (laurit) [reply instrumentation-api/src/main/jflex/SqlSanitizer.jflex]: Added a method that reads text matched by current token and strips double quotes and backticks.

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]
nice 👍

**Pull request merged**